boost.cmake

```diff
@@ -82,8 +82,37 @@ MACRO(SEARCH_FOR_BOOST)
       Boost_${UPPERCOMPONENT}_LIBRARY_RELEASE
       )
   ENDFOREACH()
+
+  # On darwin systems, we must link againt boost_python with unresolved symbols.
+  # We then remove boost_python from the global Boost_LIBRARIES list to handle it with specific care.
+  LIST(REMOVE_ITEM Boost_LIBRARIES ${Boost_PYTHON_LIBRARY})
```

Comment:
I do not think so. The only thing to do is to use macro `TARGET_LINK_BOOST_PYTHON` to link against boost_python. It will automatically properly links again boost_python according to the OS (https://github.com/boostorg/build/pull/78).